### ISSUE TITLE
[Feat] : Introducing more options for the users to SignUp

### ISSUE THREAD

**Issue by parthtyagii** [piyushgarg-dev/piyushgargdev-nextjs#214]
**Is your feature request related to a problem? Please describe.**
There are only two options available for a user to signup for the website, i.e. Discord and GitHub. 

**Describe the solution you'd like**
Two things that we can do :
1. Add our own authentication system, and
2. Adding Login with Google functionality.

**Describe alternatives you've considered**
Other alternatives could be using social media apps like Facebook and Instagram for authentication because their user base is quite high and so is the possibility of the person having an account on these social media platforms.

**Additional context**

![issueP](https://github.com/piyushgarg-dev/piyushgargdev-nextjs/assets/80752912/a2506e96-3a37-4890-a0e0-34e99c4ec5ba)

**Comment by ChavvaNarayana** [comment]
It's a great idea though what I am concerned about is this is a public website and there is no such need to log in and sign up.

If in case the owner needs this future that they may add some paid features or membership plans in the future, then it should be implemented.

Needs discussion

@TheMohit2003  @piyushgarg-dev

**Comment by parthtyagii** [comment]
Yeah, I agree there is no actual need for authentication on the website.

**Comment by GitHub-Sanjit** [comment]
Google SignIn Option is Added.
Fixes: [#1479]
